MoltenVK/MoltenVK/GPUObjects/MVKPixelFormats.mm

```diff
@@ -922,9 +922,14 @@
 	addVkFormatDesc( R64G64B64A64_SINT, Invalid, Invalid, Invalid, Invalid, 1, 1, 32, ColorFloat );
 	addVkFormatDesc( R64G64B64A64_SFLOAT, Invalid, Invalid, Invalid, Invalid, 1, 1, 32, ColorFloat );
 
+#if MVK_XCODE_15
+	addVkFormatDesc( B10G11R11_UFLOAT_PACK32, RG11B10Float, Invalid, FloatRG11B10, Invalid, 1, 1, 4, ColorFloat );
+	addVkFormatDesc( E5B9G9R9_UFLOAT_PACK32, RGB9E5Float, Invalid, FloatRGB9E5, Invalid, 1, 1, 4, ColorFloat );
+#else
 	addVkFormatDesc( B10G11R11_UFLOAT_PACK32, RG11B10Float, Invalid, Invalid, Invalid, 1, 1, 4, ColorFloat );
 	addVkFormatDesc( E5B9G9R9_UFLOAT_PACK32, RGB9E5Float, Invalid, Invalid, Invalid, 1, 1, 4, ColorFloat );
-
+#endif
+	
```

Comment:
There should probably be a block above, similar to the one for `MTLPixelFormat`, where `MTLVertexFormatRG11B10Float` and `MTLVertexFormatRGB9E5Float` are defined to `MTLVertexFormatInvalid` when `!MVK_XCODE_15`.

Reply:
so you mean this at the start of the file?
```cpp
#if !MVK_XCODE_15
#define MTLVertexFormatFloatRG11B10 MTLVertexFormatInvalid
#define MTLVertexFormatFloatRGB9E5 MTLVertexFormatInvalid
#endif
```

I thought of this, but decided it would be simpler to guard everything with a `#if MVK_XCODE_15` because otherwise I'd be giving properties to MTLVertexFormatInvalid, no?

Reply:
I imagined that you would keep the blocks you add below this inside the `MVK_XCODE_15` guards.

Reply:
> otherwise I'd be giving properties to MTLVertexFormatInvalid, no?

I don't think so. The `FloatRG11B10` and `FloatRGB9E5` would resolve in the `addVkFormatDescFull()` macro to `MTLVertexFormatInvalid`, so the effect would be the same as passing `Invalid` in the `#else` block.

But perhaps I'm missing another area where you think the redefine to `MTLVertexFormatInvalid` would cause problems?

Reply:
> I imagined that you would keep the blocks you add below this inside the MVK_XCODE_15 guards.

Why would I do that? The pixel formats `RG11B10Float` and `RGB9E5Float` have been available for ages now. The only difference is that these are now also available as vertex formats. I might also be completely misunderstanding what you're trying to say.

> But perhaps I'm missing another area where you think the redefine to MTLVertexFormatInvalid would cause problems?

I wasn't sure if the functions are guarded against giving the Invalid formats any properties. But I guess based on your comment that they are guarded and they would effectively become a no-op?

Reply:
> I wasn't sure if the functions are guarded against giving the Invalid formats any properties. But I guess based on your comment that they are guarded and they would effectively become a no-op?

The functions are not strictly guarded for (but perhaps they should be for clarity and security), but the `_mtlPixelFormatDescriptions` entry for those formats will be populated with `MTLVertexFormatInvalid`, and `buildMTLFormatMaps()` will handle that, as it does for all the other formats that are redefined as invalid.

And your calls to `addGPUOSMTLVtxFmtCaps()` are guarded by `MVK_XCODE_15` (this might have been what @cdavis5e was referring to above). Incidentally, we could probably do away with these `MVK_XCODE_15` guards around `addGPUOSMTLVtxFmtCaps()` by adding tests for `MTLVertexFormatInvalid` at the top of the `addMTLVertexFormatCapabilities()` functions, as you suggest.